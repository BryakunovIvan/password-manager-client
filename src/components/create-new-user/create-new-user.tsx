import React, { useState } from 'react';
import { addUser } from '../../db/userPublisher';

export const CreateNewUser = () => {
	const [name, setName] = useState('');
	const [secret, setSecret] = useState('');

	const handleClick = () => {
		addUser({ name, secret });
	};

	return (
		<div>
			Создайте пользователя
			<input value={name} placeholder="Введите имя" onChange={(e) => setName(e.target.value)} />
			<input value={secret} placeholder="Введите имя" onChange={(e) => setSecret(e.target.value)} />
			<button onClick={handleClick}>Создать</button>
		</div>
	);
};
