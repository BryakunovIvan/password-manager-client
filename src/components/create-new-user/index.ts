export { CreateNewUser } from './create-new-user';
