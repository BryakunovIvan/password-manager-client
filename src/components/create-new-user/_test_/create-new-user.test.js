import { render } from '@testing-library/react';
import React from 'react';
import { CreateNewUser } from '../create-new-user';

test('renders correctly', () => {
	const { container } = render(<CreateNewUser />);
	expect(container).toMatchSnapshot();
});
