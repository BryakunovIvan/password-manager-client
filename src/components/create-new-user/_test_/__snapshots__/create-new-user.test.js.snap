// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="create-user"
  >
    Создайте пользователя
    <input
      placeholder="Введите имя"
      value=""
    />
    <input
      placeholder="Введите имя"
      value=""
    />
    <button>
      Создать
    </button>
  </div>
</div>
`;
